src/schema-tools/superjson/utils.ts

```diff
@@ -33,6 +41,7 @@ export function getProviderForProfile(
     return getProvider(superJson, name);
   }
 
+  // TODO: this case should not happen - covered by super json normalization
```

Comment:
It can still happen when it's coalescing on line 37. Would be great to throw that specified profile does not exist.